src/components/SignUpDialogueBox.tsx

```diff
@@ -55,18 +120,53 @@ const SignUpDialogueBox = (props: DialogueProps) => {
     setContact(e.target.value);
   };
 
-  const onNonprofitRegister = () => {
-    // GraphQL mutation goes here for registering nonProfit
-    // attributes: username, password, mission, description, displayName, contact
-  };
-
-  const onVolunteerRegister = () => {
-    // GraphQL mutation goes here for registering volunteer
-    // attributes: username, password
+  const handleSubmit = async (e: SyntheticEvent) => {
+    e.preventDefault();
+    if (props.isVolunteer) {
+      let volunteerData = await registerVolunteer({
+        variables: {
+          username,
+          password,
+          confirmPassword,
+        },
+      });
+      if (volunteerData.data.registerVolunteer.ok) {
+        client.writeData({
+          data: {
+            volunteerID: volunteerData.data.registerVolunteer.volunteer.id,
+            token: volunteerData.data.registerVolunteer.token,
+          },
+        });
+        history.push('/volunteer-dashboard');
+      }
+    } else {
+      let nonprofitData = await registerNonprofit({
+        variables: {
+          username,
+          password,
+          confirmPassword,
+          mission,
+          description,
+          displayName,
+          contact,
+        },
+      });
+      if (nonprofitData.data.registerNonprofit.ok) {
+        client.writeData({
+          data: {
+            nonprofitID: nonprofitData.data.registerNonprofit.nonprofit.id,
+            token: nonprofitData.data.registerNonprofit.token,
+          },
+        });
+        history.push('/nonprofit-dashboard');
```

Comment:
would recommend looking into ReactRouterV5.1 it's nicer to use.  Not high priority though